app/pages/Recovery/util.ts

```diff
@@ -0,0 +1,304 @@
+    );
+    if (!credentialOnChain) {
+        return {
+            credentialIndex: undefined,
+            policy: {
+                validTo: getCurrentYearMonth(),
+                createdAt: getCurrentYearMonth(),
+                revealedAttributes: {},
+            },
+        };
+    }
+
+    return {
+        credentialIndex: parseInt(credentialOnChain[0], 10),
+        policy: credentialOnChain[1].value.contents.policy,
+    };
+}
+
+/**
+ * Attempts to recover the credential with the given credId.
+ * @param credId: credId of the credential, which is to be recovered
+ * @param blockHash: block at which the function will look up the account info
+ * @param credentialNumber: credential number of the credential on it's identity
+ * @param identityId: id of the credential's identity
+ * @returns If the credential has existed on chain, returns an object containing the credential and its accounts. If it never existed, returns undefined.
+ */
+async function recoverCredential(
+    credId: string,
+    blockHash: string,
+    credentialNumber: number,
+    identityId: number
+) {
+    const accountInfo = await getAccountInfo(credId, blockHash);
+
+    if (!accountInfo) {
+        return undefined;
+    }
+
+    const firstCredential = accountInfo.accountCredentials[0].value.contents;
+    const address = await getAddressFromCredentialId(
+        firstCredential.regId || firstCredential.credId
+    );
+
+    const { credentialIndex, policy } = getCredentialOnChain(
+        credId,
+        accountInfo
+    );
+
+    const account = createAccount(
+        identityId,
+        address,
+        AccountStatus.Confirmed,
+        undefined,
+        accountInfo.accountThreshold,
+        credentialNumber === 0
+    );
+
+    const credential = createNewCredential(
+        address,
+        credentialNumber,
+        identityId,
+        credentialIndex,
+        credId,
+        policy
+    );
+
+    return { account, credential };
+}
+
+/**
+ * Attempts to recover credentials on an existing identity.
+ * @param prfKeySeed: Seed of the prfKey of the identity.
+ * @param identityId: id of the identity.
+ * @param blockHash: block at which the function recover credentials.
+ * @param global: current global parameters.
+ * @param startingCredNumber: credentialNumber, from which to start attempting to recover credentials from.
+ * @returns Returns an object containing the list of all recovered credentials and their accounts. The length of these lists are always the same, and each account matches the credential on the same index.
+ */
+export async function recoverCredentials(
+    prfKeySeed: string,
+    identityId: number,
+    blockHash: string,
+    global: Global,
+    startingCredNumber = 0
+) {
+    const credentials = [];
+    const accounts = [];
+    let credNumber = startingCredNumber;
+    while (credNumber < maxCredentialsOnAccount) {
+        const credId = await getCredId(prfKeySeed, credNumber, global);
```

Comment:
The function `getCredId` has its first parameter as `prfKey`, but here we give it the `prfKeySeed`. Is the variable name in `rustInterface` incorrect and should in fact be `prfKeySeed`?

Reply:
Yes, it is the seed.